arch/arm/Kconfig

```diff
@@ -34,6 +34,67 @@ config ARM_CUSTOM_INTERRUPT_CONTROLLER
 	  is assumed that the custom interrupt control interface implementation
 	  assumes responsibility for handling the NVIC.
 
+config ROMSTART_RELOCATION_ROM
+	bool
+	default n
+	help
+	  Relocates the rom_start region containing the boot-vector data and
+	  irq vectors to the region specified by configurations:
+	  ROMSTART_REGION_ADDRESS and ROMSTART_REGION_SIZE
+
+	  This is useful for the Linux Remoteproc framework that uses the elf-loader
+	  such that it is able to load the correct boot-vector (contained in rom_start)
+	  into the correct memory location independent of the chosen zephyr,flash
+	  ROM region.
+
+	  Most SOCs include an alias for the boot-vector at address 0x00000000
+	  so a default which might be supported by the corresponding Linux rproc driver.
+	  If it is not, this option allows to specify the addresses into the elf file.
+
+	  In general this option should be chosen if the zephyr,flash chosen node
+	  is not placed into the boot-vector memory area
+
```

Comment:
Add the description of impaction? for example: If this item is enabled, the zephyr.bin image maybe enlarged, if only zephyr.bin is used as booting image, no need to enable it.

Reply:
@JiafeiPan Yes this is a relevant place to put it indeed, I'll provide a small update of this doc tomorrow 👍

Reply:
@JiafeiPan Commit is now reworked with explainations :)